pkg/cmd/server/bootstrappolicy/infra_sa_policy.go

```diff
@@ -634,6 +638,59 @@ func init() {
 	}
 
 	err = InfraSAs.addServiceAccount(
+		InfraPetSetControllerServiceAccountName,
+		authorizationapi.ClusterRole{
+			ObjectMeta: kapi.ObjectMeta{
+				Name: PetSetControllerRoleName,
+			},
+			Rules: []authorizationapi.PolicyRule{
+				// PetSetController.podCache.ListWatch
+				{
+					APIGroups: []string{kapi.GroupName},
+					Verbs:     sets.NewString("list", "watch"),
+					Resources: sets.NewString("pods"),
+				},
+				// PetSetController.cache.ListWatch
+				{
+					APIGroups: []string{apps.GroupName},
+					Verbs:     sets.NewString("list", "watch"),
+					Resources: sets.NewString("petsets"),
+				},
+				// PetSetController.petClient
+				{
+					APIGroups: []string{apps.GroupName},
+					Verbs:     sets.NewString("get"),
+					Resources: sets.NewString("petsets"),
+				},
+				{
+					APIGroups: []string{apps.GroupName},
+					Verbs:     sets.NewString("update"),
+					Resources: sets.NewString("petsets/status"),
+				},
+				// PetSetController.podClient
+				{
+					Verbs:     sets.NewString("get", "create", "delete", "update"),
+					Resources: sets.NewString("pods"),
+				},
+				// PetSetController.petClient (PVC)
+				// This is an escalating client and we must admission check the petset
+				{
+					Verbs:     sets.NewString("get", "create"), // future "delete"
+					Resources: sets.NewString("persistentvolumeclaims"),
```

Comment:
@eparis since petsets can create PVCs and you're trying to find a way to secure the binding, you need to make sure that this isn't an escalation path.